src/membership/components/member-actions.jsx

```diff
@@ -0,0 +1,185 @@
+import * as MemberPropTypes from '../proptypes'
+import MemberEdit from './MemberEdit'
+import ShowBarcode from './show-barcode'
+
+const Action = props => (
+  <ListItem innerDivStyle={{ paddingLeft: 60 }} {...props} />
+)
+
+let HugoAction = ({
+  attrName,
+  getMemberAttr,
+  getPath,
+  member,
+  people,
+  primaryText,
+  push
+}) => {
+  if (!getMemberAttr(member)[attrName]) return null
+  return people.size > 1 &&
+    people.some(p => p !== member && getMemberAttr(p)[attrName]) ? null : (
+    <Action
+      leftIcon={<Rocket />}
+      onClick={() => push(getPath(member.get('id')))}
+      primaryText={primaryText}
+    />
+  )
+}
+HugoAction = connect(
+  ({ user }) => ({ people: user.get('people') }),
+  { push }
+)(HugoAction)
+
+const HugoNominateAction = props => (
+  <HugoAction
+    attrName="hugo_nominator"
+    getPath={id => `/hugo/nominate/${id}`}
+    primaryText="Nominate for the Hugo Awards"
+    {...props}
+  />
+)
+
+const HugoVoteAction = props => (
+  <HugoAction
+    attrName="wsfs_member"
+    getPath={id => `/hugo/vote/${id}`}
+    primaryText="Vote for the Hugo Awards"
+    {...props}
+  />
+)
+
+const EditAction = ({ member }) => {
+  const infoStyle = { color: 'rgba(0, 0, 0, 0.870588)' }
+  const badgeName = member.get('badge_name') || member.get('preferred_name')
+  const publicName = [
+    member.get('public_first_name'),
+    member.get('public_last_name')
+  ]
+    .filter(n => n)
+    .join(' ')
+    .trim()
+  return (
+    <MemberEdit member={member}>
+      <Action
+        leftIcon={<ContentCreate style={{ top: 12 }} />}
+        primaryText="Edit personal information"
+        secondaryText={
+          <p>
+            Badge name: <span style={infoStyle}>{badgeName}</span>
+            <br />
+            Public name:{' '}
+            <span style={infoStyle}>{publicName || '[not set]'}</span>
+            <br />
+          </p>
+        }
+        secondaryTextLines={2}
+      />
+    </MemberEdit>
+  )
+}
+
+const BarcodeAction = ({ attr, member }) => {
+  return attr.badge || member.get('daypass') ? (
+    <ShowBarcode memberId={member.get('id')}>
+      <Action leftIcon={<Receipt />} primaryText="Show registration barcode" />
+    </ShowBarcode>
+  ) : null
+}
+
+let UpgradeAction = ({ member, paidPaperPubs, purchaseData, push }) => {
+  const mpt = purchaseData && purchaseData.getIn(['new_member', 'types'])
+  if (!mpt) return null
+  const prevAmount = mpt.getIn([member.get('membership'), 'amount']) || 0
+  const upgrade = mpt.some(t => t.get('amount') > prevAmount)
+  const addPP = paidPaperPubs && !member.get('paper_pubs')
+  return upgrade || addPP ? (
+    <Action
+      leftIcon={<ThumbUp style={upgrade && addPP ? { top: 12 } : null} />}
+      onClick={() => push(`/upgrade/${member.get('id')}`)}
+      primaryText={upgrade ? 'Upgrade membership' : 'Add paper publications'}
+      secondaryText={upgrade && addPP ? 'and/or add paper publications' : ''}
+    />
+  ) : null
```

Comment:
Would you consider breaking this out into 3 `<Action>` blocks to simplify the logic in the display?